### PR TITLE
Fixing self assigment bug

### PR DESCRIPTION
**Pull request by domielias**
## Description

I noticed that there was a issue that mention that there were problems with self assigment so I fixed the issue #18717 by @qarmin

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/Teletext.cpp
Comment (howie-f) [inline xbmc/video/Teletext.cpp:1288]: you need to remove this line completely as it has no effect

Comment (domielias) [reply xbmc/video/Teletext.cpp:1288]: I removed and only leave 
image

Comment (howie-f) [reply xbmc/video/Teletext.cpp:1288]: yes, but this statement `m_RenderInfo.PageChar[32+i];` does nothing and will generate a new warning on the build system. the original line also did nothing, just the self-assignment, but didn't change the value
so it's fine if you remove it entirely and just go with
```c
else
{
   SetPosX(33+i);
}
```

Comment (domielias) [reply xbmc/video/Teletext.cpp:1288]: Alright, thank you. I will push again.

Comment (domielias) [reply xbmc/video/Teletext.cpp]: Check now

**Comment by howie-f** [comment]
looks good, now it would be great if you could squash into a single commit and i would maybe call it like `[fix] remove self-assignment`. are you working with git on the command line and know how to achieve this?

**Comment by domielias** [comment]
> looks good, now it would be great if you could squash into a single commit and i would maybe call it like `[fix] remove self-assignment`. are you working with git on the command line and know how to achieve this?

I'm working in command line but I don't know so i found a command that use rebase

File: xbmc/video/Teletext.cpp
Comment (ksooo) [inline xbmc/video/Teletext.cpp:1288]: Are we sure that this is the right fix?

Comment (howie-f) [reply xbmc/video/Teletext.cpp:1288]: indeed it's not obvious to me what's the purpose of the line in its original state. just did a runtime-test without it and haven't seen regression on the teletext. what could be behind it, you have an idea? @ksooo

Comment (ksooo) [reply xbmc/video/Teletext.cpp:1288]: Sorry, no idea. But I would like to get an idea, what the real intention of the writer of that code was instead of "blindly" removing this line. If we now delete it we delete any trace that obviously here should be done more than nothing.

Comment (ksooo) [reply xbmc/video/Teletext.cpp:1288]: OMG, that code is 11 years old... but the person who comitted it is still around, @AlwinEsch would do you say? Do you remember your intentions?

Comment (domielias) [reply xbmc/video/Teletext.cpp:1288]: I notice now that this issue was posted before but was closed for not use the format. https://github.com/xbmc/xbmc/issues/15361

Comment (howie-f) [reply xbmc/video/Teletext.cpp:1288]: this is the corresponding compiler warning. let's just wait for Alwin's response. then we can go on ;)

Comment (AlwinEsch) [reply xbmc/video/Teletext.cpp:1288]: It is from the time I stole the code, unfortunately it is still there in the original.

I think it was about bringing in something for flashing, which was probably never finished.

In any case, removing it makes sense.

Comment (howie-f) [reply xbmc/video/Teletext.cpp:1288]: thanks @AlwinEsch

**Comment by AlwinEsch** [comment]
Maybe also this can be removed https://github.com/xbmc/xbmc/blob/fbb2a74c3a8a8f9ea168fde118bad24c5c02ca79/xbmc/video/Teletext.cpp#L3102, as set two times there.

**Comment by howie-f** [comment]
@domielias 

so now we can tidy up your branch, please do the following:
checkout your branch: `git checkout fix-self-assignment-bug`
then rebase it against master branch: `git rebase master`
this should return you to the two commits you had before. you can check it with `git log --oneline master..`
it should show you the two remaining commits ahead master.

to squash them into a single commit please do `git reset --soft HEAD~1` ... followed by `git commit --amend`
(..you could also squash with `rebase -i` but right now it's simpler with a soft reset)

there would be a single commit with the correct changes that you can push up then.
git can feel a bit stubborn and opaque in the beginning, but don't hesitate to ask

**Comment by howie-f** [comment]
hey @domielias :) would you like to finish this pr? the diff looks fine, we just may need it in a single commit

**Comment by utkzas** [comment]
Hello guys I am new to Open Source. I want to contribute to KODI and I have studied C++ Data Structures and Algorithms. Is it enough to grasp the working of C++ Applications like KODI? from where can I learn ?

**Review by ksooo — APPROVED** [review]

**Comment by ksooo** [comment]
@DaveTBlake this one is good to go in for v19. No risk.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by DaveTBlake** [comment]
jenkins build this please

**Comment by enen92** [comment]
jenkins build this please

**Comment by howie-f** [comment]
this doesnt build with the merge commit. needs to be squashed or rebased

**Comment by ksooo** [comment]
@domielias please squash down the 4 commits to one and we are ready to go.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
I think the PR author lost interest in this PR. As it does not contain an important fix, I remove the PR from v19.

**Comment by domielias** [comment]
Hi, sorry for my ausent. I had some personal issue. I will do the the rebase in few hours. Thanks for comment the guide @howie-f  and sorry @ksooo.

**Comment by domielias** [comment]
@howie-f I did what you said. 

When i rebase i receive the following error: 
image

So i merge the file with the current change and then later I don't see the two commit  
image

So I don't know if make a git pull from the project and then git push. 

Thanks for your help.

**Comment by howie-f** [comment]
hi @domielias 
please try a `git push -f`
the -f parameter is important, means --force and should overwrite your branch. thanks

**Comment by domielias** [comment]
@howie-f Ready, please check it.

**Review by howie-f — APPROVED** [review]
nice, this looks like we need it :+1 i'll merge once the build is complete.. thank you

File: xbmc/pvr/addons/PVRClientMenuHooks.cpp
Comment (ksooo) [inline xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: This change is unrelated. Please remove this change from this PR and open a separate PR for it.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/pvr/addons/PVRClientMenuHooks.cpp
Comment (ksooo) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: Please rebase your branch to latest upstream master and you will see that this fix already went in.

Comment (ksooo) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: Your branch is based on master from Nov 2020. ;-)

Comment (ksooo) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: `git fetch upstream`
`git rebase upstream/master`
`git push origin -ff <your-branch-name>`

Comment (howie-f) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: yes, overseen that. sorry :)

Comment (domielias) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp]: @ksooo I tried but i created two commit when i tried to remove the change in this file.

**Comment by howie-f** [comment]
okay, now we're very close to finish.
first you need to setup your upstream repository:
`git remote add upstream https://github.com/xbmc/xbmc.git`

next you can rebase your branch on upstream-master:
`git rebase upstream/master`

now you squash down to one commit:
`git reset --soft HEAD~1`
`git commit --amend --no-edit`

finally force-push:
`git push -f`

**Comment by domielias** [comment]
@howie-f Check it now

File: xbmc/pvr/addons/PVRClientMenuHooks.cpp
Comment (howie-f) [inline xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: rebase was successful, but this change is still wrong.
please now
`git checkout HEAD^ -- xbmc/pvr/addons/PVRClientMenuHooks.cpp`
`git commit --amend --no-edit`
`git push -f`

**Review by howie-f — CHANGES_REQUESTED** [review]

File: xbmc/pvr/addons/PVRClientMenuHooks.cpp
Comment (domielias) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: @howie-f I did it. 
image

Comment (howie-f) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: i'm sorry, `git commit -a --amend --no-edit` is correct.
i missed to write -a

Comment (domielias) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: You're right, sorry. 
I missed to write -a.

Comment (howie-f) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: sorry, but the change is still there...
please once again the three commands in a row.. it should work

Comment (domielias) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: I continue having four files changed.
image

Comment (howie-f) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: tried it on your repo:
```
fhw:~/domi/xbmc (fix-self-assigment-bug) $ git checkout HEAD^ -- xbmc/pvr/addons/PVRClientMenuHooks.cpp
fhw:~/domi/xbmc (fix-self-assigment-bug) $ git commit -a --amend --no-edit
[fix-self-assigment-bug 637ef3a5bf] [fix] remove self-assigment
 Author: Elias Rodriguez <45079270+domielias@users.noreply.github.com>
 Date: Wed Nov 4 14:25:17 2020 -0400
 3 files changed, 3 insertions(+), 4 deletions(-)
```
can it be that you need to `cd D:\Open Source Project\PullRequest` before you issue the first command?

Comment (howie-f) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp:29]: maybe windows cannot understand HEAD^, please try this instead
`git checkout  4a937d9 -- xbmc/pvr/addons/PVRClientMenuHooks.cpp`
`git commit -a --amend --no-edit`

Comment (domielias) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp]: Yes, Windows don't recognize the HEAD. That work

Comment (howie-f) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp]: juhuuuu, then push -f

Comment (domielias) [reply xbmc/pvr/addons/PVRClientMenuHooks.cpp]: I did it.

**Review by ksooo — APPROVED** [review]
Finally! :-)

**Review by howie-f — APPROVED** [review]
thanks, finally we got it
jenkins build this please

**Comment by howie-f** [comment]
jenkins build this please

File: xbmc/network/websocket/WebSocketManager.cpp
Comment (howie-f) [inline xbmc/network/websocket/WebSocketManager.cpp:52]: @domielias there 2 `;;` at the end of this line. can you please remove one of them and then
`git commit -a --amend --no-edit` and a force-push?
noticed it in the diff here https://jenkins.kodi.tv/job/BuildMulti-PR/18460/artifact/PR18725.diff

Comment (domielias) [reply xbmc/network/websocket/WebSocketManager.cpp]: @howie-f Sure, i made it. Check it now.

Comment (howie-f) [reply xbmc/network/websocket/WebSocketManager.cpp]: :+1:

**Comment by howie-f** [comment]
jenkins build this please

**Comment by DaveTBlake** [comment]
That was epic, thanks all.

**Pull request merged**
